Snippets/Sqs/Sqs_6/Usage.cs

```diff
@@ -0,0 +1,391 @@
+using Amazon.SimpleNotificationService;
+using Amazon.SQS;
+using Amazon.SQS.Model;
+using NServiceBus;
+using NServiceBus.Pipeline;
+
+
+class Usage
+{
+    void AssumePermissionsOnPolicy(EndpointConfiguration config)
+    {
+        #region assume-permissions
+
+        var transport = new SqsTransport();
+        
+        transport.Policies.SetupTopicPoliciesWhenSubscribing = false;
+        
+        config.UseTransport(transport);
+
+        #endregion
+    }
+
+    void WildcardsAccount(EndpointConfiguration config)
+    {
+        #region wildcard-account-condition
+
+        var transport = new SqsTransport();
+
+        transport.Policies.AccountCondition = true;
+
+        config.UseTransport(transport);
+
+        #endregion
+    }
+
+    void WildcardsPrefix(EndpointConfiguration config)
+    {
+        #region wildcard-prefix-condition
+
+        var transport = new SqsTransport();
+
+        transport.Policies.TopicNamePrefixCondition = true;
+
+        config.UseTransport(transport);
+
+        #endregion
+    }
+
+    void WildcardsNamespace(EndpointConfiguration config)
+    {
+        #region wildcard-namespace-condition
+
+        var transport = new SqsTransport();
+
+        transport.Policies.TopicNamespaceConditions
+            .Add("Sales.");
+        transport.Policies.TopicNamespaceConditions
+            .Add("Shipping.HighValueOrders.");
+
+        config.UseTransport(transport);
+
+        #endregion
+    }
+
+    Usage(EndpointConfiguration endpointConfiguration)
+    {
+        #region SqsTransport
+
+        var transport = new SqsTransport();
+        // S3 bucket only required for messages larger than 256KB
+        transport.S3 = new S3Settings("myBucketName", "my/key/prefix");
+
+        endpointConfiguration.UseTransport(transport);
+
+        #endregion
+    }
+
+    void DelayedDelivery(EndpointConfiguration endpointConfiguration)
+    {
+        #region DelayedDelivery
+
+        
+
+        #endregion
+    }
+
+    void CredentialSource(EndpointConfiguration endpointConfiguration)
+    {
+        #region CredentialSource
+
+        var transport = new SqsTransport(
+            new AmazonSQSClient(new InstanceProfileAWSCredentials()), 
+            new AmazonSimpleNotificationServiceClient());
+
+        endpointConfiguration.UseTransport(transport);
+
+        #endregion
+    }
+
+    void S3CredentialSource(EndpointConfiguration endpointConfiguration, string bucketName, string keyPrefix)
+    {
+        #region S3CredentialSource
+
+        var transport = new SqsTransport();
+
+        transport.S3 = new S3Settings(bucketName, keyPrefix, 
+            new AmazonS3Client(new InstanceProfileAWSCredentials()));
+
+        endpointConfiguration.UseTransport(transport);
+
+        #endregion
+    }
+
+    void MaxTTL(EndpointConfiguration endpointConfiguration)
+    {
+        #region MaxTTL
+
+        var transport = new SqsTransport();
+
+        transport.MaxTimeToLive = TimeSpan.FromDays(10);
```

Comment:
in MSMQ I've been using object initializers for these snippets (as that seems to be the code style we should follow now...): https://docs.particular.net/transports/msmq/transportconfig?version=msmqtransport_2